### PR TITLE
feature variant-buttons

### LINKED ISSUE #51: Create the Button Default Component
## description

implement a button for default use with styled-components

## where

`src/components/Button`

## expect use

- basic
```jsx
<Button>Text</Button>
```

This button needs to contain the default, hover, active, focus, disabled styles

## preview

image

## figma 

https://www.figma.com/file/YeJgCxq0y9yrK5Tk2r4ywF/desktop?node-id=85%3A26

### PR DESCRIPTION
**Pull request by diegocosta-dev**
resolves #51 #52 #53 #54 #55

### CODE REVIEW

**Review by davidcostadev — CHANGES_REQUESTED** [review]

File: package.json [hunk 1]
```diff
@@ -51,6 +51,7 @@
     "@apollo/react-testing": "4.0.0",
     "@types/react": "^16.9.19",
     "apollo-link-http": "^1.5.17",
+    "css-variables": "^0.1.7",
```
Comment (davidcostadev) [inline package.json:54]: This is not needed you can remove it

File: src/components/Button.tsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        variation === 'default' ? `var(--button-border-active-${variation})` : 'none'};
+  }
+
+  &:focus {
+    background-color: ${({ variation = 'default' }) =>
+      `var(--button-background-color-focus-${variation})`};
+    border: solid 2px
+      ${({ variation = 'default' }) =>
+        variation === 'default' ? `var(--button-border-focus-${variation})` : 'none'};
+    box-shadow: 0px 0px 0px 2px
```
Comment (davidcostadev) [inline src/components/Button.tsx:42]: 0px -> 0

File: src/components/Button.tsx [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+      ${({ variation = 'default' }) => `var(--button-box-shadow-focus-${variation})`};
+  }
+
+  &:disabled {
+    color: ${({ variation = 'default' }) => `var(--button-color-disabled-${variation})`};
+    background-color: ${({ variation = 'default' }) =>
+      `var(--button-background-color-disabled-${variation})`};
+    border: solid 2px
```
Comment (davidcostadev) [inline src/components/Button.tsx:50]: border-color: {color}

File: src/components/Button.tsx [hunk 4]
```diff
@@ -0,0 +1,56 @@
+import styled from 'styled-components';
+
+type TextProps = {
+  variation?: Variations;
+  rouded?: boolean;
```
Comment (davidcostadev) [inline src/components/Button.tsx:5]: typo: rouded -> rounded

File: src/components/Button.tsx [hunk 5]
```diff
@@ -0,0 +1,56 @@
+import styled from 'styled-components';
+
+type TextProps = {
```
Comment (davidcostadev) [inline src/components/Button.tsx:3]: ButtonProps or Props

File: src/components/Button.tsx [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+      `var(--button-background-color-hover-${variation})`};
+    border: solid 2px
+      ${({ variation = 'default' }) =>
+        variation === 'default' ? `var(--button-border-hover-${variation})` : 'none'};
+  }
+
+  &:active {
+    background-color: ${({ variation = 'default' }) =>
+      variation === 'default' ? `var(--button-background-color-active-${variation})` : 'none'};
+    border: solid 2px
```
Comment (davidcostadev) [inline src/components/Button.tsx:31]: border-color: {color}

File: src/components/Button.tsx [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+  border: solid 2px ${({ variation = 'default' }) => `var(--button-border-${variation})`};
+  padding: 11px 13px;
+  border-radius: ${({ rouded }) => (rouded ? '37px' : '5px')};
+  cursor: ${({ disabled }) => (disabled ? '' : 'pointer')};
+  outline: none;
+
+  &:hover {
+    background-color: ${({ variation = 'default' }) =>
+      `var(--button-background-color-hover-${variation})`};
+    border: solid 2px
```
Comment (davidcostadev) [inline src/components/Button.tsx:23]: border-color: {color}

File: src/components/Button.tsx [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+  rouded?: boolean;
+};
+
+type Variations = 'primary' | 'danger' | 'light';
+
+const Button = styled.button<TextProps>`
+  color: ${({ variation = 'default' }) => `var(--button-color-${variation})`};
+  background-color: ${({ variation = 'default' }) => `var(--button-background-color-${variation})`};
+  font-size: 16px;
+  border: solid 2px ${({ variation = 'default' }) => `var(--button-border-${variation})`};
```
Comment (davidcostadev) [inline src/components/Button.tsx:14]: border: {width} {style} {color}

File: src/components/Button.tsx [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+  padding: 11px 13px;
```
Comment (davidcostadev) [inline src/components/Button.tsx:15]: Please use pair numbers

File: src/components/Button.tsx [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+  variation?: Variations;
+  rouded?: boolean;
+};
+
+type Variations = 'primary' | 'danger' | 'light';
+
+const Button = styled.button<TextProps>`
+  color: ${({ variation = 'default' }) => `var(--button-color-${variation})`};
+  background-color: ${({ variation = 'default' }) => `var(--button-background-color-${variation})`};
+  font-size: 16px;
```
Comment (davidcostadev) [inline src/components/Button.tsx:13]: Please use vars on font-sizes

**Review by davidcostadev — CHANGES_REQUESTED** [review]

**Review by gustavoFreireS — APPROVED** [review]

**Review by davidcostadev — APPROVED** [review]

**Pull request merged**